cloud/scope/util.go

```diff
@@ -26,6 +26,9 @@ const (
 
 // GetNsgNamesFromId returns the names of the NSGs with the provided IDs
 func GetNsgNamesFromId(ids []string, nsgs []*infrastructurev1beta2.NSG) []string {
+	if len(ids) == 0 {
```

Comment:
when there are no NSGs defined in the template, the comparison should be with nil array